src/QMCHamiltonians/CoulombPBCAA.cpp

```diff
@@ -436,11 +440,41 @@ CoulombPBCAA::Return_t CoulombPBCAA::evalConsts(bool report)
 {
   mRealType Consts = 0.0; // constant term
   mRealType v1;           //single particle energy
+  mRealType vl_r0 = AA->evaluateLR_r0(); // short-range Madelung background
+  mRealType vs_k0 = AA->evaluateSR_k0(); // long-range Madelung background
+  if (Quasi2D) // background term has z dependence
+  { // just evaluate the Madelung term
+    for (int ispec=1; ispec<NumSpecies; ispec++)
+      if (Zspec[ispec] != Zspec[0])
+        throw std::runtime_error("Quasi2D assumes same charge");
+    if (report)
+    {
+      app_log() << "    vlr(r->0) = " << vl_r0 << std::endl;
+      app_log() << "   1/V vsr_k0 = " << vs_k0 << std::endl;
+    }
+    // make sure we can ignore the short-range Madelung sum
+    mRealType Rws = Ps.getLattice().WignerSeitzRadius;
+    mRealType rvsr_at_image = Rws*AA->evaluate(Rws, 1.0/Rws);
+    if (rvsr_at_image > 1e-6)
```

Comment:
I don't have an example where the default Ewald `alpha = sqrt(kc/(2*rc))` fails this condition.
This is just a defensive check in case someone runs into the weird case in the future.